api/v1alpha1/addon_types.go

```diff
@@ -40,111 +36,38 @@ type Addons struct {
 	CSIProviders *CSI `json:"csi,omitempty"`
 }
 
-func (Addons) VariableSchema() clusterv1.VariableSchema {
-	return clusterv1.VariableSchema{
-		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
-			Description: "Cluster configuration",
-			Type:        "object",
-			Properties: map[string]clusterv1.JSONSchemaProps{
-				"cni":               CNI{}.VariableSchema().OpenAPIV3Schema,
-				"nfd":               NFD{}.VariableSchema().OpenAPIV3Schema,
-				"clusterAutoscaler": ClusterAutoscaler{}.VariableSchema().OpenAPIV3Schema,
-				"csi":               CSI{}.VariableSchema().OpenAPIV3Schema,
-				"ccm":               CCM{}.VariableSchema().OpenAPIV3Schema,
-			},
-		},
-	}
-}
-
 type AddonStrategy string
 
 // CNI required for providing CNI configuration.
 type CNI struct {
-	// +optional
-	Provider string `json:"provider,omitempty"`
-	// +optional
-	Strategy AddonStrategy `json:"strategy,omitempty"`
-}
-
-func (CNI) VariableSchema() clusterv1.VariableSchema {
-	supportedCNIProviders := []string{CNIProviderCalico, CNIProviderCilium}
-
-	return clusterv1.VariableSchema{
-		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
-			Type: "object",
-			Properties: map[string]clusterv1.JSONSchemaProps{
-				"provider": {
-					Description: "CNI provider to deploy",
-					Type:        "string",
-					Enum:        variables.MustMarshalValuesToEnumJSON(supportedCNIProviders...),
-				},
-				"strategy": {
-					Description: "Addon strategy used to deploy the CNI provider to the workload cluster",
-					Type:        "string",
-					Enum: variables.MustMarshalValuesToEnumJSON(
-						AddonStrategyClusterResourceSet,
-						AddonStrategyHelmAddon,
-					),
-				},
-			},
-			Required: []string{"provider", "strategy"},
-		},
-	}
+	// CNI provider to deploy.
+	// +kubebuilder:validation:Enum=Calico;Cilium
+	Provider string `json:"provider"`
+	// Addon strategy used to deploy the CNI provider to the workload cluster.
+	// +kubebuilder:validation:Enum=ClusterResourceSet;HelmAddon
+	Strategy AddonStrategy `json:"strategy"`
```

Comment:
this was optional before and now its mandatory to provide?

Reply:
It's always been mandatory, hasn't it? At least enforced in handlers, e.g. https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/537 - this is just ensuring it's set in variable definition.

Honestly I would like to make it optional and default to `HelmAddon` tbh (or in fact I'd like to remove `ClusterResourceSet` completely considering it's deprecated, but we need it for now).